setup.py

```diff
@@ -57,6 +67,26 @@ def is_requirement(line):
     )
 
 
+def get_edx_platform_version():
+    """
+    Returns the pip package version for Open-edX
+    """
+    file_name = 'open_edx_version.txt'
+    command = 'pip show Open-edX > {}'.format(file_name)
+    subprocess.call(command, shell=True)
+    version = 0.0
+
+    with open(file_name, 'r') as file:
+        lines = file.readlines()
+        for line in lines:
+            if line.startswith('Version'):
+                name, version = line.split(': ')
+                version = float(version)
```

Comment:
Please add a try-catch, to stop the process, if the version cannot be converted to float. @andrey-canon

Reply:
the process is stooped, if the version cannot be converted to float because this raises an exception

Reply:
Same comment, sorry, my mistake, we can't stop all the process, we should log the error and let the process move ahead. @andrey-canon

Reply:
if the default case is an error, it's better raise the exception to show to the final user what it's exactly happening

Reply:
Stop all the deploy process just to realize that we are using the incorrect version, if you log the error and don't stop the process, you can show to the user what it's exactly happening. @andrey-canon

Reply:
Why don't stop the process when the default case stops the process ???? suppress an exception and then raise another with less information

Reply:
Anywise, stop the process but, Why lees information? with the right information... RuntimeError("Unable to get the edX platform version while running {command}"), that's all.

Reply:
This is the same error  without and with try-except, imo the first one give more information about the error 
```
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/edx/src/openedx-proversity-reports/setup.py", line 115, in <module>
        install_requires=load_requirements(REQUIREMENTS_FOLDER_PATH),
      File "/edx/src/openedx-proversity-reports/setup.py", line 38, in load_requirements
        edx_platform_version = get_edx_platform_version()
      File "/edx/src/openedx-proversity-reports/setup.py", line 95, in get_edx_platform_version
        version = float("UNKNOWN")
    ValueError: could not convert string to float: UNKNOWN
```
```
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/edx/src/openedx-proversity-reports/setup.py", line 118, in <module>
        install_requires=load_requirements(REQUIREMENTS_FOLDER_PATH),
      File "/edx/src/openedx-proversity-reports/setup.py", line 38, in load_requirements
        edx_platform_version = get_edx_platform_version()
      File "/edx/src/openedx-proversity-reports/setup.py", line 98, in get_edx_platform_version
        raise RuntimeError("Unable to get the edX platform version")
    RuntimeError: Unable to get the edX platform version
```

Reply:
RuntimeError(Unable to get the edX platform version, could not convert string to float: UNKNOWN). @andrey-canon

Reply:
Even better, we can create a custom exception class to manage this kind of the erros.

Reply:
do you want to suppress an exception that gives the main reason  and create another one to show a custom message ?

Reply:
No, append to the custom message, the reason...

class MyException():

    message = 'Unable to get the edX platform version'

    def __init__(self, message):
        print('{}: {}'.format(self.message, message))

Reply:
All of this, is in order to manage the exception, and not just "fail". @andrey-canon

Reply:
that looks as redundant code